web/src/installers/index.ts

```diff
@@ -1199,6 +1199,17 @@ export class Installer {
         return {error: {message: "Rook 1.0.4 is not compatible with Kubernetes 1.20+"}};
       }
     }
+
+    if (this.spec.prometheus && this.spec.prometheus.version && this.spec.prometheus.serviceType) {
+      if (this.spec.prometheus.serviceType != "" && this.spec.prometheus.serviceType != "ClusterIP" && this.spec.prometheus.serviceType != "NodePort") {
+        return {error: {message: `Supported Prometheus service types are "NodePort" and "ClusterIP", not "${this.spec.prometheus.serviceType}"`}};
```

Comment:
I think `this.spec.prometheus.serviceType` needs to be escaped unless that's handled at a higher level

Reply:
Actually I think this is fine

Reply:
I think that's handled, but I can try it in a test

Reply:
tried, worked fine